### DISCUSSION TITLE
Feature request : first order low pass filter

### DISCUSSION THREAD

**Discussion by Tristan107** [psambit9791/jDSP#78] [Ideas]
How would I implement a first order low pass filter using JDSP ? Is it possible yet ?

Also is it possible to apply a bilinear transform using JDSP ? (I'm  talking about his in Matlab : https://fr.mathworks.com/help/signal/ref/bilinear.html) This would be a requirement to implement a first order low pass filter.

**Comment by psambit9791** [comment]
Hi @Tristan107 

I am unsure what you're asking for. The library provides different IIR Filter options - Butterworth, Bessel, Chebyshev.. All of them allow you to set an `order` when you are calling the `lowPassFilter()` function.

Currently, JDSP does not support Bilinear transform. I am unsure how this would be a requirement for a first order low pass filter.

Can you please elaborate.

**Reply by Tristan107** [reply]
I've been told that a "first order" filter is not the same as a specific filter with order = 1. "first order" filter would be a more basic filter than a Butterworth or Bessel filter, and it would be implemented using a bilinear transformation.

It may look like this : https://dsp.stackexchange.com/questions/11345/digital-implementation-of-first-order-analog-filter-using-bilinear-transformatio
